### PR TITLE
Xiao Xiao, 1st round zoo

### PR DESCRIPTION
**Pull request by xiaoxiao9689**
Xiao Xiao, 1st round zoo upload, (two classical codes: random codes and Justesen codes)

### CODE REVIEW

File: codes/classical/bits/random_code.yml [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+#   text...
+#   more text...
+protection: 'The code can protect against random errors, e.g. from Binary Symmetry Channel (BSC), discrete memoryless channel.'
+
+
+
+
+# This field starts a list of specific labeled subfields; do not leave it
+# empty. If empty, comment out. Also, indentations are important!
+features: 'Given N and the rate R, the minimum distance of a typical random code (TRC) is given by the Gilbert–Varshamov (GV) bound d=N\delta_{GV}(2R).
```
Comment (AlexanderBarg) [inline codes/classical/bits/random_code.yml:50]: We need to define the function \delta_{GV}, namely \delta_{GV}(x):= h^{-1}(1-x), 0\le x \le 1,  and \delta_{GV}(x)=0 if for all other x.

File: codes/classical/bits/random_code.yml [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+realizations:
+  # List and explain the different "domains" of realizations in list items.
+  # Group closely related realizations into a single item. It's OK to have a
+  # single list item. Each item should more or less correspond to what would be
+  # said in a paragraph.
+  - 'Distributed Storage Systems \cite{doi: 10.1109/TPDS.2008.251}'
+
+notes:
+    - 'Code features refer to the asymptotic limit, i.e. The code approach the minimum distance with probability 1 as N goes to infiity.'
```
Comment (AlexanderBarg) [inline codes/classical/bits/random_code.yml:114]: "The code approach the minimum distance with probability 1 as N goes to infiity." 
-> 
"We consider ensembles of codes of increasing length $n$ and cardinality $M=nR$, and argue about the limiting parameters of the typical or average codes in the ensemble as $n\to\infty$."

File: codes/classical/q-ary_digits/justesen.yml [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+
+# code id, physical, logical are all lower case
+code_id: justesen
+physical: q-ary_digits
+logical: q-ary_digits
+
+# Apostrophes are denoted by two apostrophe characters, i.e., ''
+# Code title (SINGULAR) + first reference(s) (optional).
+name: 'Justesen_code'
+introduced: '\cite{manual:{MacWilliams/Sloane, The Theory of Error-Correcting Codes, Sec.10.11}}'
```
Comment (AlexanderBarg) [inline codes/classical/q-ary_digits/justesen.yml:27]: The correct reference is J. Justesen's paper of about 1972 on "A class of constructive asymptotically good codes" (I am citing the title from memory). They are indeed presented in the cited book, among other places.

File: codes/classical/q-ary_digits/justesen.yml [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+
+# Anything applicable to a larger parent set of codes (see below) should go in
+# that entry instead of here.
+description: 'Given m, Justesen_code is a binary code over GF(2^m), with length n=2mN, dimension k=mK, where (N=2^m-1,K) is the reed-solomon outer code.
+              The code can be considered as concatenation of a Reed-Solomon outer code with N distinct inner codes.'
```
Comment (AlexanderBarg) [inline codes/classical/q-ary_digits/justesen.yml:32]: The code can be considered as concatenation of a Reed-Solomon outer code **of length N over F_{2^m}** with N distinct inner codes.'

File: codes/classical/q-ary_digits/justesen.yml [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+logical: q-ary_digits
+
+# Apostrophes are denoted by two apostrophe characters, i.e., ''
+# Code title (SINGULAR) + first reference(s) (optional).
+name: 'Justesen_code'
+introduced: '\cite{manual:{MacWilliams/Sloane, The Theory of Error-Correcting Codes, Sec.10.11}}'
+
+# Anything applicable to a larger parent set of codes (see below) should go in
+# that entry instead of here.
+description: 'Given m, Justesen_code is a binary code over GF(2^m), with length n=2mN, dimension k=mK, where (N=2^m-1,K) is the reed-solomon outer code.
```
Comment (AlexanderBarg) [inline codes/classical/q-ary_digits/justesen.yml:31]: "where (N=2^m-1,K) is the reed-solomon outer code."
Replace with "where the (N=2^m-1,K) are the parameters of the Reed-Solomon outer code **over F_{2^m}.**


"Justesen_code is a binary code over GF(2^m)" 
This is incorrect: a binary code is over F_2 (not F_{2^m}), and Justesen's codes are indeed binary.

File: codes/classical/q-ary_digits/justesen.yml [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+
+realizations: ''
+  # List and explain the different "domains" of realizations in list items.
+  # Group closely related realizations into a single item. It's OK to have a
+  # single list item. Each item should more or less correspond to what would be
+  # said in a paragraph.
+  # - 'Code used in DVDs \cite{doi:####...}, 5G, etc.'
+  # - 'Realized in trapped-ion quantum devices \cite{arXiv:####.#####}, etc.'
+
+notes: 'The code can be improved and extend the range of R from 0 to 1 by puncturiing the code (by erasing s digits from each inner code), resulting in a code of length n=(2m-s)N, rate R=mk/(2m-s)N. The lower bound of the distance of the punctured code
```
Comment (AlexanderBarg) [inline codes/classical/q-ary_digits/justesen.yml:105]: (by erasing s digits from each inner **code word**)

File: codes/classical/q-ary_digits/justesen.yml [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+        approaches d=(1-R/r)H^{-1}(1-R) as m goes to infinity, where r is the solution to R=r^2/(1+log(1-H^{-1}(1-r))), and H being the binary entropy function.'
+relations:
+  parents:
+    - code_id: reed_solomon
+      detail: 'The smallest code family that includes this code that is defined over the same physical space structure or alphabet.'
+    - code_id: concatenated code
+  cousins:
```
Comment (AlexanderBarg) [inline codes/classical/q-ary_digits/justesen.yml:112]: concatenated codes

File: codes/classical/q-ary_digits/justesen.yml [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+realizations: ''
+  # List and explain the different "domains" of realizations in list items.
+  # Group closely related realizations into a single item. It's OK to have a
+  # single list item. Each item should more or less correspond to what would be
+  # said in a paragraph.
+  # - 'Code used in DVDs \cite{doi:####...}, 5G, etc.'
+  # - 'Realized in trapped-ion quantum devices \cite{arXiv:####.#####}, etc.'
+
+notes: 'The code can be improved and extend the range of R from 0 to 1 by puncturiing the code (by erasing s digits from each inner code), resulting in a code of length n=(2m-s)N, rate R=mk/(2m-s)N. The lower bound of the distance of the punctured code
+        approaches d=(1-R/r)H^{-1}(1-R) as m goes to infinity, where r is the solution to R=r^2/(1+log(1-H^{-1}(1-r))), and H being the binary entropy function.'
```
Comment (AlexanderBarg) [inline codes/classical/q-ary_digits/justesen.yml:106]: Line 106 describes an intermediate result. Please replace the code parameters with the parameters cited in MacWilliams and Sloane, Thm.10.12, page 214.

File: codes/classical/properties/random.yml [hunk 9]
```diff
@@ -6,15 +6,43 @@
 code_id: random
 
 name: 'Random code'
-#introduced: ''
+introduced: '\cite{doi: 10.1109/TIT.2002.800480}'
 
-description: 'Code whose construction is non-deterministic in some way, i.e., codes that utilize an elements of randomness somewhere in their construction. Members of this class range from fully non-deterministic codes, to codes whose multi-step construction is deterministic with the exception of a single step.'
+description: 'Code whose construction is non-deterministic in some way, i.e., codes that utilize an elements of randomness somewhere in their construction. Members of this class range from fully non-deterministic codes, to codes whose multi-step construction is deterministic with the exception of a single step.
```
Comment (valbert4) [inline codes/classical/properties/random.yml:11]: Surround latex math with \(...\)

Comment (valbert4) [inline codes/classical/properties/random.yml:11]: Can do
`description: |
       paragraph 1

       par 2...`

File: codes/classical/properties/random.yml [hunk 10]
```diff
@@ -6,15 +6,43 @@
...
+              Typically, random codes are selected with uniform distribution from a ensemble of codes. Examples are random binary codes and random binary linear codes.
+              A random binary code is a set of 2^{K}=2^{NR} codewords with length N. Each bit in the codeword is randomly chosen between 0 and 1 with equal probability.
+              A random binary linear code is generated from a random chosen K generators of length N where each bit of the generators is randomly chosen between 0 and 1 with equal probability.
+              Equivalently, A random binary linear code is defined by a randomly generated K by N generator matrix where each entry is randomly chosen between 0 and 1 with equal probability.'
 
 #notes:
 #  - ''
 
+protection: 'The code can protect against random errors, e.g. from Binary Symmetry Channel (BSC), discrete memoryless channel.'
+
```
Comment (valbert4) [inline codes/classical/properties/random.yml:21]: rate: field, describing Shannon argument to achieving capacity with random codes

File: codes/classical/properties/random.yml [hunk 11]
```diff
@@ -6,15 +6,43 @@
...
+features:            
+  decoders:
+    - 'Ball-Collision Decoding \cite{doi:10.1007/978-3-642-22792-9_42}'
+    - 'Finiasz and Sendrier (FS-ISD Decoding)\cite{doi:10.1007/978-3-642-10366-7_6}'
+             
+realizations:
+    - 'Distributed Storage Systems \cite{doi: 10.1109/TPDS.2008.251}'
+  
+notes:
+    - 'Given N and the rate R, the minimum distance of a typical random code (TRC) is given by the Gilbert–Varshamov (GV) bound d=N\delta_{GV}(2R).
```
Comment (valbert4) [inline codes/classical/properties/random.yml:31]: First note --> protection field, adding defn of h^{-1}. Second note can be merged into rate, although more specific random-code-related info is needed there.

**Pull request closed without merging**